app/src/ereferrals/java/org/eyeseetea/malariacare/data/database/datasources/CSVFileDataSource.java

```diff
@@ -0,0 +1,114 @@
+package org.eyeseetea.malariacare.data.database.datasources;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+import org.eyeseetea.malariacare.data.database.model.AnswerDB;
+import org.eyeseetea.malariacare.data.database.model.HeaderDB;
+import org.eyeseetea.malariacare.data.database.model.MatchDB;
+import org.eyeseetea.malariacare.data.database.model.OptionAttributeDB;
+import org.eyeseetea.malariacare.data.database.model.OptionDB;
+import org.eyeseetea.malariacare.data.database.model.ProgramDB;
+import org.eyeseetea.malariacare.data.database.model.QuestionDB;
+import org.eyeseetea.malariacare.data.database.model.QuestionOptionDB;
+import org.eyeseetea.malariacare.data.database.model.QuestionRelationDB;
+import org.eyeseetea.malariacare.data.database.model.QuestionThresholdDB;
+import org.eyeseetea.malariacare.data.database.model.SurveyDB;
+import org.eyeseetea.malariacare.data.database.model.TabDB;
+import org.eyeseetea.malariacare.data.database.utils.Session;
+import org.eyeseetea.malariacare.data.database.utils.populatedb.PopulateDB;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ICSVFileRepository;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+
+public class CSVFileDataSource implements ICSVFileRepository {
```

Comment:
Repositories and Datasources have the goal to the  domain abstraction of data layer and get and save domain entities but download csv process is not invoked from domain and is not entities, then DataSources and repositories concepts are wrong in this process, create interfaces for abstract is not necessary in this case because generate an unnecessary noise and complexity.
I think that we need only a MetadataAPIClient (github client) and new class that coordinate the all proccess including check local and remote csv version, invoke the download of every csv and save on local